umbraco/Umbraco-CMS
# Use IFileService to save macro partial views in package migration

### Prerequisites


### Description
When a package migration tried to save macro partial views from an embedded package.xml file (like in The Starter Kit), it directly used `File.WriteAllText()` instead of the `IFileService.SavePartialViewMacro()`. If the `Views/MacroPartials` directory didn't exist yet, this also caused a `DirectoryNotFoundException` to be thrown (which would cause a `BootFailedException` when the migrations are automatically run).

Testing steps:
1. Before applying this PR on a clean install, remove the `Views/MacroPartials` directory and try to install The Starter Kit (`dotnet add package Umbraco.TheStarterKit --version 9.0.0-rc002`): this will throw an exception while booting up;
2. Manually create the `Views/MacroPartials` directory and start the site: the package migration will now successfully run and the `FeaturedProducts.cshtml` and `LatestBlogposts.cshtml` files should be created in this directory;
3. Apply this PR to a clean install and repeat the first step: it should now run the package migration without issues and the macro partial view files should also exist.
